src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -218,6 +218,12 @@ def _on_acquire(self, _):
         # the acquisition is started
         self._tab_data.main.is_acquiring.value = True
 
+        # there is a race condition on the is_acquiring/_on_acquisition callback
+        # if the stream is playing, the callback can start, and then the acquisition starts
+        # the callback will pause the stream, and the acquisition will timeout
+        # so we pause the stream synchronously here
+        self._tab.streambar_controller.pauseStreams()
```

Comment:
Indeed, that's needed here. The reason is that `is_acquiring.value = True` calls the `on_acquisition` callback, but as the callback is "@call_in_wx_main()` protected, it will only be run *after* the current function is done running (as it's called by the GUI event loop). Eventually, the acquisition starts in a separate thread, and it might or not start the first stream before `on_acquisition` is called.

So, you should also *remove* the call to `pauseStreams()` in `on_acquisition`, line 203, to be entirely sure it will behave correctly.  That function should only do display-related changes (ie, disable/enable buttons).